pkg/kubelet/kubelet.go

```diff
@@ -1313,6 +1323,19 @@ func (kl *Kubelet) GenerateRunContainerOptions(pod *api.Pod, container *api.Cont
 	if err != nil {
 		return nil, err
 	}
+
+	// If required, create special mounts to order cAdvisor to gather custom metrics.
+	if kl.enableCustomMetrics {
+		customMetricsMount, err := custommetrics.MakeCustomMetricsMountIfRequired(
+			pod, container, kl.getPodContainerDir(pod.UID, container.Name), kl.maxCustomMetricsPerPod)
+		if err != nil {
```

Comment:
Great thanks for spotting this! This is rather not desire behavior. I've added verification of custom metrics to canRunPod function. It should help.